### PR TITLE
make logbook implementations optional 

### PR DESCRIPTION
**Pull request by shroffk**
With this change... site specific products will have to explicitly list the logbook implementations they want to use.
so nsls2, ess, and frib can include olog and olog-es while fhi can include the elog one.

the common product will still build and include all which can be good for people using the common product.

Finally, it will be logbook_factory preference which will still be required to select from the included logbook implementations.

### CODE REVIEW

**Comment by shroffk** [comment]
This PR must wait till #1875 is resolved (sorry)

**Review by tanviash — APPROVED** [review]

**Comment by kingspride** [comment]
I emailed Kay and Kunal about this logbook factory issue, because I was not able to set the elog via my settings.ini.

I hope to narrow down why its conflicting with the olog app, but I am not a java "native speaker"

**Review by kasemir — APPROVED** [review]

File: core/logbook/src/main/resources/logbook_preferences.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# ------------------------------
+# Package org.phoebus.logbook.ui
```
Comment (kasemir) [inline core/logbook/src/main/resources/logbook_preferences.properties:2]: I think this is typo, should say "org.phoebus.logbook" without .ui

Comment (shroffk) [reply core/logbook/src/main/resources/logbook_preferences.properties]: yes this was a typo...thanks for catching it

**Comment by shroffk** [comment]
@kasemir @tanviash @georgweiss 
I think the new preference and its location is more appropriate the how we had it in the past.
The logbook client selection preference is now part of core...

Unfortunately, this might require some changes in your site specific preferences to match the new package name

**Comment by kasemir** [comment]
OK on changing the preference name from my side.

**Review by georgweiss — APPROVED** [review]

File: core/logbook/src/main/java/org/phoebus/logbook/LogbookPreferences.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package org.phoebus.logbook;
+
+import org.phoebus.framework.preferences.AnnotatedPreferences;
+import org.phoebus.framework.preferences.Preference;
+import org.phoebus.framework.preferences.PreferencesReader;
+
+import java.util.logging.Level;
+
+import static org.phoebus.ui.application.PhoebusApplication.logger;
```
[the remark this answers is not in the document]
Comment (kasemir) [reply core/logbook/src/main/java/org/phoebus/logbook/LogbookPreferences.java:9]: core-logbook's LogService already has its own logger, so it really makes sense that core-logbook doesn't use the PhoebusApplication logger.

**Pull request merged**